trust-quorum/src/connection_manager.rs

```diff
@@ -446,19 +446,16 @@ impl ConnMgr {
             );
 
             // Inform the main task that accepted connection is established
-            if let Err(e) = main_tx
-                .send(ConnToMainMsg {
-                    task_id: task::id(),
-                    msg: ConnToMainMsgInner::Accepted {
-                        addr,
-                        peer_id: baseboard_id,
-                    },
-                })
-                .await
-            {
-                // The system is shutting down
+            if let Err(_) = main_tx.try_send(ConnToMainMsg {
+                task_id: task::id(),
+                msg: ConnToMainMsgInner::Accepted {
+                    addr,
+                    peer_id: baseboard_id,
+                },
+            }) {
+                // The system is shutting down or we've overloaded the main channel
```

Comment:
I think that shedding load like this here seems like a reasonable approach --- if we can't tell the main task that we have accepted a new connection, we just give up on it and close the TCP connection. But, this makes me wonder whether the overall TQ protocol is resilient to peers shedding load like this due to (probably transient) overload conditions. Will the peer be able to either try to talk to us again or find a different node to talk to? And, should we consider sending some kind of "hi, i'm shedding load" type message first so that the peer can distinguish between the "temporarily hit channel capacity" and "i am shutting down, don't talk to me again" cases?